lib/dor/shelve.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Dor
+  # Push file changes for shelve-able files into the stacks
+  # NOTE: ShelvingService would have been an ideal name, but it collided with the
+  #       same in dor-services
+  class Shelve
+    # @param work [Dor::Item] the item to transfer to Stacks
+    def self.push(work)
+      new(work).shelve
+    end
+
+    # @param work [Dor::Item] the item to transfer to Stacks
+    def initialize(work)
+      @work = work
+    end
+
+    def shelve
+      # retrieve the differences between the current contentMetadata and the previously ingested version
+      diff = shelve_diff
+      stacks_object_pathname = stacks_location
+      # determine the location of the object's files in the stacks area
+      stacks_druid = DruidTools::StacksDruid.new work.id, stacks_object_pathname
+      stacks_object_pathname = Pathname(stacks_druid.path)
+      # determine the location of the object's content files in the workspace area
+      workspace_druid = DruidTools::Druid.new(work.id, Config.stacks.local_workspace_root)
+      workspace_content_pathname = workspace_content_dir(diff, workspace_druid)
+      # delete, rename, or copy files to the stacks area
+      DigitalStacksService.remove_from_stacks(stacks_object_pathname, diff)
+      DigitalStacksService.rename_in_stacks(stacks_object_pathname, diff)
```

Comment:
This was code copied straight out of dor-services without changes.   So I'm just preserving the weirdness for this PR at least.

This is also why I needed to mess with rubocop, because this code was using Ruby 2.5 stuff, but the common-accessioning's rubocop was set to use Ruby 2.2 parser.  I tried to keep the changes separated into different commits. I think that's the best I can do without copying code in and changing it at the same time.

Reply:
Ahh, missed the separate commits.